### DISCUSSION TITLE
How to build NDK r21e for aarch64 architecture?

### DISCUSSION THREAD

**Discussion by diskria** [android/ndk#1885] [Q&A] [answered]
Hi everyone. Help me please with building NDK r21e (required version for project, I can't upgrade it bcz I make a fork) for aarch64 architecture. 

I really need it for fun: I've been wanting to build an application in Termux that I'm developing on a desktop computer for a long time. this has become a dream for a long time and there is no point in it, I just want to understand how development tools are created. When M1 processors appeared, I thought this would become possible, and if I understand everything correctly, this has already happened in the NDK r23c update (https://github.com/android/ndk/wiki/Changelog-r23#r23c). But the problem is that I need exactly the r21e version. Relatively few changes have passed between these revisions (probably), so I think it's easy, at least possible. Is that right? 

Maybe you can show what changes need to be made, or at least show links to the necessary changes in the code? As I understand it, something needs to be updated in toolchains. Maybe there is some guide for adding aarch64 support for NDK earlier than r23c?

**Comment by DanAlbert** [comment, marked as answer]
The only thing we can offer is `git log`. You can probably tell by how long that bug took to resolve that it's not going to be easy.

**Reply by diskria** [reply]
ok, I'll try to figure how to work in google git. there is an unusual navigation for me, which makes me poorly understand where to start, but I'll try

**Reply by DanAlbert** [reply]
It will almost certainly be easier for you to upgrade past r21 than it will be to maintain your own for of r21. That's only going to get more difficult over time.
